### DISCUSSION TITLE
CPM - Overwrite location of Directory.Packages.props location

### DISCUSSION THREAD

**Discussion by avivanoff** [NuGet/Home#11906] [Q&A] [answered]
Is there a way to overwrite the location of Directory.Packages.props when using CPM? I tried DirectoryPackagesPropsPath and CentralPackgesFile, but no luck.

**Comment by jeffkl** [comment, marked as answer]
`DirectoryPackagesPropsPath` is set and imported very early in an MSBuild project evaluation.  There are only two ways to set it properly for an SDK-style project, use [`Directory.Build.props`](https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2022#directorybuildprops-example) or use [explicit SDK imports](https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-use-project-sdk):

```xml
<!-- Project element has SDK attribute removed, it is imported later -->
<Project>
  <PropertyGroup>
    <!-- Setting DirectoryPackagesPropsPath before anything else is imported -->
    <DirectoryPackagesPropsPath>..\..\MyCustomPackages.props</DirectoryPackagesPropsPath>
  </PropertyGroup>

  <!-- Explicitly import Microsoft.NET.Sdk top import Sdk.props -->
  <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.props" />

  <PropertyGroup>
    <!-- Normal project properties and items -->
    <TargetFramework>net6.0</TargetFramework>
  </PropertyGroup>

  <!-- Explicitly import Microsoft.NET.Sdk bottom import Sdk.targets -->
  <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" />
</Project>
```

I recommend using `Directory.Build.props` so that every project doesn't need the custom logic.  I also recommend just using the default file name `Directory.Packages.props` so that users who work in your repository that are used to NuGet's conventions will know which file is used.

**Reply by avivanoff** [reply]
We had been using common props and targets file long before Directory.Build.props/targets were introduced, and we still use them even with SDK-style projects, so this approach naturally fits with our build setup.

As a side note, I personally consider Directory.Build.props/targets and SDK-style projects not flexible enough, and always use explicit SDK imports and common props/targets.
